com.unity.render-pipelines.universal/Runtime/Passes/DrawObjectsPass.cs

```diff
@@ -92,11 +92,39 @@ public override void Execute(ScriptableRenderContext context, ref RenderingData
                 }
                 #endif
 
-                context.DrawRenderers(renderingData.cullResults, ref drawSettings, ref filterSettings, ref m_RenderStateBlock);
+                var sceneOverrideMode = DebugDisplaySettings.Instance.renderingSettings.sceneOverrides;
```

Comment:
We should probably encapsulate part of this logic in a helper debug class  or a manager. There's too much implementation details here to add debug to a custom pass.

Reply:
Should now be sorted - we need to be able to render these contexts multiple times for certain modes (wireframe specifically) so there's now an enumerable that we iterate over and a class "DebugRenderSetup" that deals with setting/modifiying whatever data we need